code/controllers/stonedmc/subsystem/evacuation.dm

```diff
@@ -0,0 +1,228 @@
+	if(evac_status != EVACUATION_STATUS_INITIATING)
+		return FALSE
+	GLOB.enter_allowed = TRUE
+	evac_time = null
+	evac_status = EVACUATION_STATUS_STANDING_BY
+	command_announcement.Announce("Evacuation has been cancelled.", "Priority Alert", new_sound='sound/AI/evacuate_cancelled.ogg')
+	var/datum/shuttle/ferry/marine/evacuation_pod/P
+	for(var/i = 1 to MAIN_SHIP_ESCAPE_POD_NUMBER)
+		P = shuttle_controller.shuttles["[MAIN_SHIP_NAME] Evac [i]"]
+		P.toggle_ready()
+	return TRUE
+
+
+/datum/controller/subsystem/evacuation/proc/get_status_panel_eta()
+	switch(evac_status)
+		if(EVACUATION_STATUS_INITIATING)
+			var/eta = EVACUATION_ESTIMATE_DEPARTURE
+			. = "[(eta / 60) % 60]:[add_zero(num2text(eta % 60), 2)]"
+		if(EVACUATION_STATUS_IN_PROGRESS)
+			. = "NOW"
+
+/datum/controller/subsystem/evacuation/proc/announce_evac_completion()
+	command_announcement.Announce("ATTENTION: Evacuation complete.", "Priority Alert", new_sound='sound/AI/evacuation_complete.ogg')
+	evac_status = EVACUATION_STATUS_COMPLETE
+
+
+/datum/controller/subsystem/evacuation/proc/enable_self_destruct(override)
+	if(dest_status != NUKE_EXPLOSION_INACTIVE)
+		return FALSE
+	if(!override && flags_scuttle & (FLAGS_SELF_DESTRUCT_DENY|FLAGS_SDEVAC_TIMELOCK))
+		return FALSE
+	dest_status = NUKE_EXPLOSION_ACTIVE
+	dest_master.toggle()
+	set_security_level(SEC_LEVEL_DELTA)
+	return TRUE
+
+
+/datum/controller/subsystem/evacuation/proc/cancel_self_destruct(override)
+	if(dest_status != NUKE_EXPLOSION_ACTIVE)
+		return FALSE
+	var/obj/machinery/self_destruct/rod/I
+	var/i
+	for(i in SSevacuation.dest_rods)
+		I = i
+		if(I.active_state == SELF_DESTRUCT_MACHINE_ARMED && !override)
+			dest_master.state("<span class='warning'>WARNING: Unable to cancel detonation. Please disarm all control rods.</span>")
+			return FALSE
+
+	dest_status = NUKE_EXPLOSION_INACTIVE
+	for(i in dest_rods)
+		I = i
+		if(I.active_state == SELF_DESTRUCT_MACHINE_ACTIVE || (I.active_state == SELF_DESTRUCT_MACHINE_ARMED && override)) I.toggle(1)
+	dest_master.toggle(1)
+	dest_index = 1
+	command_announcement.Announce("The emergency destruct system has been deactivated.", "Priority Alert", new_sound='sound/AI/selfdestruct_deactivated.ogg')
+	if(evac_status == EVACUATION_STATUS_STANDING_BY)
+		set_security_level(SEC_LEVEL_RED, TRUE)
+	return TRUE
+
+
+/datum/controller/subsystem/evacuation/proc/initiate_self_destruct(override)
+	if(dest_status >= NUKE_EXPLOSION_IN_PROGRESS)
+		return FALSE
+
+	var/obj/machinery/self_destruct/rod/I
+	for(var/i in dest_rods)
+		I = i
+		if(I.active_state != SELF_DESTRUCT_MACHINE_ARMED && !override)
+			dest_master.state("<span class='warning'>WARNING: Unable to trigger detonation. Please arm all control rods.</span>")
+			return FALSE
+	command_announcement.Announce("DANGER. DANGER. Self destruct system activated. DANGER. DANGER. Self destruct in progress. DANGER. DANGER.", "Priority Alert")
+	trigger_self_destruct(,,override)
```

Comment:
```suggestion
	trigger_self_destruct(override = override)
```
Named arguments are a little less performant, but this is not a hot proc and code clarity is more important here.